src/adr/0040-generalize-event-bus.md

```diff
@@ -0,0 +1,27 @@
+#40. Generalize EventBus
+
+Date: 2020-06-11
+
+## Status
+
+Proposed
+
+## Context
+
+When searching for users by quota ratio if we set the value of the parameters to 0, the search feature is supposed to return newly created users
+who have not received any email yet at that point. However, this is not the case due to the quotas are currently being initialized only after
+an user receive the first email.
+
+In order to solve this problem, we need to initialize user quotas upon creation time. Event driven system allows indirection so that the users api 
+don't need a dependency on the quota components. The EventBus components currently defined in in mailbox-api would be suitable for this purpose.
+
+## Decision
```

Comment:
I suggest:

For quota-search to initialize a given user and remove it at the right time while keeping that feature as a plugin, we decided to adopt the Event Driven pattern we already use in mailbox and create new events related to user management (UserCreated, UserRemoved and so on).

To achieve that, we will extract the EventBus api out of the mailbox-api and make both mailbox-api and data-api depend on that new module eventbus-api.